pkg/object/unstructured.go

```diff
@@ -88,3 +112,58 @@ func GetCRDGroupKind(u *unstructured.Unstructured) (schema.GroupKind, bool) {
 	}
 	return emptyGroupKind, false
 }
+
+// UnknownTypeError captures information about a type for which no information
+// could be found in the cluster or among the known CRDs.
+type UnknownTypeError struct {
+	GroupKind schema.GroupKind
+}
+
+func (e *UnknownTypeError) Error() string {
```

Comment:
So I agree in general that we should use values instead pointers when possible. But errors are usually implemented as pointers, even the errors produced with `fmt.Errorf` are pointers. Most patterns and example around error handling (like using `errors.Is` and `errors.As`) assume the error is a pointer, so I think it would be surprising to users if we just return a value.